rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template

```diff
@@ -380,7 +380,9 @@ static message_type_support_callbacks_t callbacks = {
   "@(spec.base_type.type)",
   &register_type__@(spec.base_type.type),
   &publish__@(spec.base_type.type),
-  &take__@(spec.base_type.type)
+  &take__@(spec.base_type.type),
+  nullptr,  // convert ros to dds (handled differently for C++)
```

Comment:
In order to keep `rmw_opensplice_cpp` code from needing to know which typesupport it has (C or C++), I'd propose we leave this as is. @jacquelinekay do you have another idea of how to implement this? Other thoughts?